### PR TITLE
Add a layer for the bundle config manifest

### PR DESCRIPTION
**Pull request by rumpl**
This layer is needed for hub to be able to show the image tag and not
only the invocation image tag.

### CODE REVIEW

File: converter/types.go [hunk 1]
```diff
@@ -112,6 +119,13 @@ func prepareOCIBundleConfig(mediaType string) bundleConfigPreparer {
 func prepareNonOCIBundleConfig(blob []byte) (*PreparedBundleConfig, error) {
 	man, err := schema2.FromStruct(schema2.Manifest{
 		Versioned: schema2.SchemaVersion,
+		Layers: []distribution.Descriptor{
```
Comment (silvin-lubecki) [inline converter/types.go]: I think you can factorize with the Config descriptor just below 👍

Comment (rumpl) [reply converter/types.go]: Done

File: remotes/mocks_test.go [hunk 2]
```diff
@@ -37,15 +37,31 @@ func (r *mockResolver) Pusher(_ context.Context, ref string) (remotes.Pusher, er
...
+		returnErrorValues: ret,
+		idx:               0,
+	}
 }
 
 func (p *mockPusher) Push(ctx context.Context, d ocischemav1.Descriptor) (content.Writer, error) {
 	p.pushedDescriptors = append(p.pushedDescriptors, d)
 	buf := &bytes.Buffer{}
 	p.buffers = append(p.buffers, buf)
+	err := error(nil)
```
Comment (silvin-lubecki) [inline remotes/mocks_test.go]: `var err error` instead?

Comment (rumpl) [reply remotes/mocks_test.go]: Done

File: remotes/mocks_test.go [hunk 3]
```diff
@@ -37,15 +37,31 @@ func (r *mockResolver) Pusher(_ context.Context, ref string) (remotes.Pusher, er
...
 }
 
 func (p *mockPusher) Push(ctx context.Context, d ocischemav1.Descriptor) (content.Writer, error) {
 	p.pushedDescriptors = append(p.pushedDescriptors, d)
 	buf := &bytes.Buffer{}
 	p.buffers = append(p.buffers, buf)
+	var err error
+	if p.returnErrorValues != nil {
+		err = p.returnErrorValues[p.idx]
+		p.idx = p.idx + 1
```
Comment (silvin-lubecki) [inline remotes/mocks_test.go]: Maybe you could pop the first error from the list and get rid of `p.idx`?
```go
if len(p.returnErrorValues) > 0 {
    err = p.returnErrorValues[0]
    p.returnErrorValues = p.returnErrorValues[1:]
}
```

File: converter/types.go [hunk 4]
```diff
@@ -109,14 +109,22 @@ func prepareOCIBundleConfig(mediaType string) bundleConfigPreparer {
...
+
 func prepareNonOCIBundleConfig(blob []byte) (*PreparedBundleConfig, error) {
+	desc := nonOCIDescriptorOf(blob)
 	man, err := schema2.FromStruct(schema2.Manifest{
 		Versioned: schema2.SchemaVersion,
-		Config: distribution.Descriptor{
-			MediaType: schema2.MediaTypeImageConfig,
-			Size:      int64(len(blob)),
-			Digest:    digest.FromBytes(blob),
+		Layers: []distribution.Descriptor{
```
Comment (silvin-lubecki) [inline converter/types.go]: Maybe add a comment here to explain why we are actually adding a layer here?

Comment (rumpl) [reply converter/types.go]: Done

**Pull request merged**